pkgs/flojoy/flojoy/env_var.py

```diff
@@ -0,0 +1,72 @@
+import sys
+from keyrings.cryptfile.cryptfile import CryptFileKeyring
+import keyring
+from typing import Optional
+from pathlib import Path
+from .CONSTANTS import FLOJOY_DIR, CREDENTIAL_FILE, KEYRING_KEY
+import os
+from .config import logger
+
+__all__ = [
+    "get_env_var",
+    "set_env_var",
+    "delete_env_var",
+]
+
+
+def get_keyring():
+    if sys.platform.lower() == "linux":
+        kr = CryptFileKeyring()
+        kr.keyring_key = KEYRING_KEY
+        keyring.set_keyring(kr)
+    return keyring
+
+
+def get_env_var(key: str) -> Optional[str]:
+    kr = get_keyring()
+    return kr.get_password("flojoy", key)
+
+
+def set_env_var(key: str, value: str):
+    kr = get_keyring()
+    kr.set_password("flojoy", key, value)
+    home = str(Path.home())
+    file_path = os.path.join(home, os.path.join(FLOJOY_DIR, CREDENTIAL_FILE))
+
+    if not os.path.exists(file_path):
+        logger.info(f"{file_path} does not exist")
+        with open(file_path, "w") as f:
+            f.write(key)
+        logger.info(f"Env var written to {file_path}")
+        return
+
+    logger.info(f"{file_path} exists, writing env to {file_path}")
+    with open(file_path, "r") as f:
+        keys = f.read().strip().split(",")
+        if key not in keys:
+            keys.append(key)
+
+    with open(file_path, "w") as f:
+        f.write(",".join(keys))
+
+
+def delete_env_var(key: str):
+    home = str(Path.home())
+    file_path = os.path.join(home, os.path.join(FLOJOY_DIR, CREDENTIAL_FILE))
+
+    if not os.path.exists(file_path):
+        return
+
+    with open(file_path, "r") as f:
+        keys = f.read().strip().split(",")
+
+    if key not in keys:
+        return
+
+    keys.remove(key)
+
+    with open(file_path, "w") as f:
+        f.write(",".join(keys))
+
+    kr = get_keyring()
+    kr.delete_password("flojoy", key)
```

Comment:
Since we're only ever reading the variables from the system keyring, do we even still need to write it to the credentials file? I don't think anything actually reads from that file :thinking:

Reply:
They are use to retrieve the variable in get .../env/ route. Not sure with the route don't read it directly from keyring ? :/
I can remove it and just use Keyring

Reply:
Oh, it's so the app can know all the env vars the user has set. Their is no "get_all_passwords" from Keyright. One need to know the name of the key to retrieve. But the current implementation is not the most secure

Reply:
@39bytes @LatentDream The purpose of writing to the credential file was to store the environment keys set by the user. This enables us to display them on the Environment modal, ensuring users are aware if a secret for this environment has already been set. I think we can keep it as keyring doesn't allow to list all keys...

Reply:
Agreed:)